File: week3/eunbeen/피로도.java
import java.util.*;

class Solution {
    public boolean[] check = {};
    public int answer = -1;
    public int solution(int k, int[][] dungeons) {
        check = new boolean[dungeons.length];
        getAdventure(dungeons, k, 0);
        
        return answer < 0 ? 0 : answer;
    }
    private void getAdventure(int[][] dungeons, int k, int count){
        for(int i=0; i<dungeons.length; i++){
            if(!check[i] && dungeons[i][0] <= k){
                check[i] = true;
                getAdventure(dungeons, k -dungeons[i][1], count+1);
                check[i] = false;
            }
        }
        
        if (answer < count){
            answer = count;
        }
    }
}

File: week3/lee/피로도.py
from itertools import *

def solution(k, dungeons):
    answer = -1
    for each in list(permutations(dungeons,len(dungeons))): # 라이브러리로 모든 경우의 수 추출
        temp = k
        count = 0
        for req, piro in each:
            if(temp >= req):
                temp -= piro
                count += 1
        answer = max(answer, count)

    return answer
print(solution(80, [[80,20],[50,40],[30,10]]	))